### DISCUSSION TITLE
seaweedfs behind rclone?

### DISCUSSION THREAD

**Discussion by gymnae** [seaweedfs/seaweedfs#4897] [Q&A]
I recently moved 4TB of data from a WebDav share to Backblaze B2. Mostly I'm using it for nextcloud, to access a library of differently sized files.

I'm mounting the object storage via rclone to a server and access the files like a normal FS, making the backend transparent to the consuming apps. All files are encrypted, since I am using the crypt wrapper of rclone to access the B2/S3 storage. 

I fiddled a lot with the settings for rclone to minimize class c transactions, but still always create costs. Downloads are no problem, since I put Cloudflare CDN in between. 
But I want to lower class C transactions, such as s3_list_objects

Could I add seaweedfs in between rclone and consumers, like Nextcloud, to benefit from the metadata storage to minimize class C transactions? Then I would not have to move storage again. 

Using the native S3 connetor from Nextcloud is not an option because it doesn't get updates that often and Nextcloud itself is not the most stable. If I lose the Nextcloud database, I lose my files, since they are not stored like on a normal file system

**Comment by chrislusf** [comment]
How much class A B C transaction costs?

Yes. SeaweedFS should be able to help. Seems the cloud drive feature should work for you.

**Reply by gymnae** [reply]
Currently, it's all in testing, but I do ramp up $0.5 per day just for class C. That may appear as not much, but it's for private use only and currently just used by me.

This is the report for October so far: 
- s3_get_object 15,630 
- s3_head_object 127,598
- b2_list_buckets 34
- b2_list_file_versions 2,082  
- s3_copy_object 4,891
-  s3_list_objects 299,596

**Reply by gymnae** [reply]
Cloud drive feature looks interesting. Does this create a metadata database to learn about "folders"? Or, simpler, avoid having to poll Backblaze for existing objects and folders?

If I were to use the cache of seaweedfs, I would disable the VFS caching of rclone and just have it act as the middle man between Backblaze and SeaweedFS.

What I like about this potential approach is that I can keep the existing rclone setup and add to it, instead of migrating, which would entail downloading all 4TB and then reuploading with the native S3 connector of seaweedfs

**Reply by gymnae** [reply]
How would this work? Can I connect seaweedfs directly to rclone, or would I need to have rclone mount B2/S3 to a local folder and add this to seaweedfs as source?

**Reply by chrislusf** [reply]
SeaweedFS FUSE => SeaweedFS cluster => B2/S3

**Reply by gymnae** [reply]
So: local user (nextcloud) -> SeaweedFS Fuse -> SeaweedFS Cluster -> rclone -> Backblaze?

**Comment by kmlebedev** [comment]
>  I am using the crypt wrapper of rclone to access the B2/S3 storage.

Perhaps in the future seaweeds will be able to encrypt itself using the aws s3crypt standard with KMS, as rclone does https://github.com/rclone/rclone/issues/7192
